### DISCUSSION TITLE
build fail

### DISCUSSION THREAD

**Discussion by phpsxg** [CoderLine/alphaTab#663] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

pageage.json run start error:
/usr/local/bin/node /usr/local/lib/node_modules/npm/bin/npm-cli.js test --scripts-prepend-node-path=auto

> @coderline/alphatab@1.3.0 test /Users/mac/sxg/work/alphaTab
> npm run generate-typescript && tsc --project tsconfig.json && concurrently --kill-others "tsc --project tsconfig.json -w" "karma start karma.conf.js --browsers Chrome --no-single-run --reporters spec,kjhtml"

> @coderline/alphatab@1.3.0 generate-typescript /Users/mac/sxg/work/alphaTab
> rimraf src/generated && ts-node --project tsconfig.build-csharp.json src.compiler/typescript/AlphaTabGenerator.ts --project tsconfig.build-csharp.json

/Users/mac/sxg/work/alphaTab/node_modules/ts-node/src/index.ts:513
    return new TSError(diagnosticText, diagnosticCodes)
           ^
TSError: ⨯ Unable to compile TypeScript:
src.compiler/typescript/CloneEmitter.ts:99:41 - error TS2345: Argument of type 'string | NodeArray<JSDocComment>' is not assignable to parameter of type 'string | MemberName'.
  Type 'NodeArray<JSDocComment>' is not assignable to type 'string | MemberName'.
    Type 'NodeArray<JSDocComment>' is not assignable to type 'string'.

99                                         collectionAddMethod
                                           ~~~~~~~~~~~~~~~~~~~

    at createTSError (/Users/mac/sxg/work/alphaTab/node_modules/ts-node/src/index.ts:513:12)
    at reportTSError (/Users/mac/sxg/work/alphaTab/node_modules/ts-node/src/index.ts:517:19)
    at getOutput (/Users/mac/sxg/work/alphaTab/node_modules/ts-node/src/index.ts:752:36)
    at Object.compile (/Users/mac/sxg/work/alphaTab/node_modules/ts-node/src/index.ts:968:32)
    at Module.m._compile (/Users/mac/sxg/work/alphaTab/node_modules/ts-node/src/index.ts:1056:42)
    at Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Object.require.extensions.(anonymous function) [as .ts] (/Users/mac/sxg/work/alphaTab/node_modules/ts-node/src/index.ts:1059:12)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @coderline/alphatab@1.3.0 generate-typescript: `rimraf src/generated && ts-node --project tsconfig.build-csharp.json src.compiler/typescript/AlphaTabGenerator.ts --project tsconfig.build-csharp.json`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @coderline/alphatab@1.3.0 generate-typescript script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/mac/.npm/_logs/2021-09-23T03_37_43_554Z-debug.log
npm ERR! Test failed.  See above for more details.

Process finished with exit code 1
[2021-09-23T03_37_43_554Z-debug.log](https://github.com/CoderLine/alphaTab/files/7214915/2021-09-23T03_37_43_554Z-debug.log)


### Expected Behavior

ok

### Steps To Reproduce

1. git clone https://github.com/CoderLine/alphaTab.git
2. npm install
3. npm run generate-typescript && tsc --project tsconfig.json && concurrently --kill-others \"tsc --project tsconfig.json -w\" \"karma start karma.conf.js --browsers Chrome --no-single-run --reporters spec,kjhtml\"
  

### Link to jsFiddle, CodePen, Project

_No response_

### Found in Version

1.3-alpha

### Platform

Other

### Environment

```markdown
- **OS**:mac
- **Browser**:chrom
- **.net Version**:2.1.818
```


### Anything else?

_No response_

**Comment by Danielku15** [comment, marked as answer]
Converted to a discussion. It is not a bug in the library that you cannot compile it locally. Can you double check which exact typescript version is used on your system? We use raw NPM in alphaTab and not yarn. And the package-lock.json points to typescript 4.2.4 which should be compiling. I could imagine that yarn installs for some reason a different version. 

You could also try rather using raw NPM instead of yarn.
